### ISSUE TITLE
Creating a bundle after babel 6 transpilation cannot find runtime dependencies

### ISSUE THREAD

**Issue by Kraxxis** [systemjs/builder#622]
I am attempting to use systemjs-builder to create a JavaScript bundle in the simplest way possible, with the least amount of dependencies. No jspm, no build system, just simple cli commands.

Here is my `package.json` devDependencies:
```javascript
 "devDependencies": {
    "babel-cli": "^6.10.1",
    "babel-plugin-transform-es2015-modules-systemjs": "^6.9.0",
    "babel-plugin-transform-runtime": "^6.9.0",
    "babel-preset-es2015": "^6.9.0",
    "systemjs-builder": "^0.15.22"
}, 
```

My `.babelrc`:
```javascript
{
    "presets": ["es2015"],
    "plugins": [
        "transform-runtime",
        "transform-es2015-modules-systemjs"
    ]
} 
```

At this point, I execute the following command to tranform my ES6 JavaScript like so:

`node ./node_modules/babel-cli/bin/babel.js --out-dir ./out/stage1/ src/`

This results in my transformed code sitting in `out/stage1`.  The JavaScript no longer has import statements, but have been transformed into SystemJS modules, with dependencies to the babel runtime like so:
```javascript
System.register(["babel-runtime/helpers/classCallCheck", "babel-runtime/helpers/createClass"], function (_export, _context) {

/// Content omitted for brevity

});
```

Next I create my builder file like so:
`systemjs.build.js`
```javascript
var Builder = require("systemjs-builder");
var builder = new Builder("./out/stage1", "./systemjs.config.js");
builder
    .bundle('index.js', '../../out/stage2/index.js')
    .then(function(){   
        console.log("done"); 
    });  
```
And my config file like so:
`systemjs.config.js`
```javascript
System.config({
    defaultJSExtensions: true,
    "map": {
        'babel-runtime/*': './node_modules/babel-runtime/*'
    }
});
```

Executing `node ./systemjs.build.js` results in the following error:

```
Unhandled rejection Error on fetch for core-js/library/fn/object/define-property.js at file:///home/krax/devel/js/try1/out/stage1/core-js/library/fn/object/define-property.js
        Loading babel-runtime/core-js/object/define-property
        Loading babel-runtime/helpers/createClass
        Loading index.js
        Error: ENOENT: no such file or directory, open '/home/krax/devel/js/try1/out/stage1/core-js/library/fn/object/define-property.js'
    at Error (native)
```

When walking the stack trace, just as the error displays, reveals a require call of the form `require("core-js/library/fn/object/define-property.js');`

If I add the configuration `"core-js/*":"./node_modules/core-js/*"` to the `map` object of the systemjs configuration, I get the same error.

If I remove the `transform-runtime` from my babeljs transformation, then the build can work, however this is unacceptable, as my project will have many files eventually.

**Comment by Kraxxis** [comment]
After poking at this for a bit longer, I have found a working solution by creating a symlink to core-js like so:

`ln -s ../../node_modules/core-js out/stage1/core-js`

However, this isnt exactly ideal.  

This appears to show that when encountering `require(...)` calls to other modules when walking the babel-runtime dependends, along with specifying a base path, results in system js not walking potential dependency paths in the current directory's `node_modules` folder.

Am I not creating my SystemJS configuration correctly?  Is there some way to properly create a map to core-js allowing systemjs to bundle properly?

**Comment by Kraxxis** [comment]
I had to add `core-js/*' : './node_modules/core-js/*'` to the `'paths'` config in my systemjs configuration.

I have a working build now. Sorry for the trouble.

**Comment by guybedford** [comment]
Sure, thanks for the update. For this use case it can be better to use a
'core-js' map config instead of paths.

**Comment by Kraxxis** [comment]
using a core-js configuration in the 'map' config was ineffective; this only worked once I had placed the configuration in the 'paths' config

**Comment by guybedford** [comment]
Did you definitely try -

map: {
  'core-js': './node_modules/corejs'
}

?

**Comment by Kraxxis** [comment]
> Did you definitely try -
>
> map: {
>   'core-js': './node_modules/corejs'
> }
> 
> ?

Yes that definitely didn't work. Both with and without your typo (corejs -> core-js).

I was able to get this working for a bit, Only to take another 4-5 steps forward and run into a similar issue again.

This time the error occurred here with the following message:

```
{ Error on fetch for core-js/library/fn/symbol.js at file:///.../my-folder/node_modules/core-js/library/fn/symbol.js
        Loading babel-runtime/core-js/symbol
        Loading babel-runtime/helpers/typeof
        Loading babel-runtime/helpers/inherits
        Loading project-name/main-screen.js
        Loading index.js
        Error: ENOENT: no such file or directory, open '/.../my-folder/node_modules/core-js/library/fn/symbol.js'
    at Error (native)
  originalErr: 
   { Error: ENOENT: no such file or directory, open '/.../my-folder/node_modules/core-js/library/fn/symbol.js'
       at Error (native)
     errno: -2,
     code: 'ENOENT',
     syscall: 'open',
     path: '.../my-folder/node_modules/core-js/library/fn/symbol.js' } }
```
The reason being is because `node_modules/core-js/library/fn/symbol` is a folder containing an `index.js` file.  

Adding the paths line ` 'core-js/library/fn/symbol': './node_modules/core-js/library/fn/symbol/index.js',` results in the systemjs bundle building, but attempting to execute the bundle results in runtime errors, requesting files from `/core/modules/es6.symbol.js`

It is clear to me I am doing something wrong or attempting something not supported.  My goal is simple: Build my src with BabelJS using the `transform-runtime` plugin, then bundle with SystemJS. But I cannot bundle the `babel-runtime/*` files with my bundle.

Do you have any recommendations?

**Comment by Kraxxis** [comment]
Some more info in my attempt to bundle `babel-runtime` and `core-js` with my project:

The reason why I can build, but the build requests more modules (like `/core-js/modules/es6.symbol.js`) is because of how the CommonJS modules are structured in `core-js/library/symbol`.  

+   The file `core-js/library/fn/symbol/index.js` uses relative pathing to its own dependencies: `../../modules/es6.symbol.js`.  The builder will include the correct file (`node_modules/core-js/library/modules/es6.symbol.js`) and its dependencies in its output. 
+   However, at runtime, because the file `core-js/library/fn/symbol/index.js` will be included as `core-js/library/fn/symbol`, the relative pathing of this files dependencies resolves to a different path (`core-js/modules/es6.symbol.js`).

In addition when I've been playing with `map` and `paths` configuration, I have seen some confusing behavior because the paths for two of the npm dependencies have overlap:

+   npm_modules/babel-runtime/**core-js**
+   npm_modules/**core-js**

its seem that resolving some module at `node_modules/babel-runtime/core-js/~~~`, sometimes gets mapped to look at `node_modules/core-js/~~~`

Is there some regex like syntax when declaring a `map` or `path` so I can say something like `'^core-js/*':'./node_modules/core-js/*'`?

Do you have any suggestions on how one can bundle `babel-runtime`?

**Comment by guybedford** [comment]
jspm was designed to handle these compatibilities automatically via `jspm install npm:babel-runtime`, perhaps try using it here.

Otherwise you want to load an already-built version of babel-runtime with SystemJS.

**Comment by Kraxxis** [comment]
I took a look at jspm, and what it does when installing `babel-runtime`, and how jspm modifies the `config.js` file.  I am still not interested in incorporating yet another package manager into my project.  I have a hard enough time keeping my dependency list slim and under control.

I saw some undocumented syntax for map config (a map key can take an object for a value? what does that do?), but i also saw how jspm solves the CommonJS  spec resolution issue i mentioned with `index.js`.

Because of this, im going to scratch using the `transform-runtime` until `systemjs-builder` can better handle CommonJS path resolution.

If I could request or suggest a change:

The CommonJS module resolution as defined by the npm docs show that when resolving `foo/bar/baz` and `baz` is a directory, and `baz/index.js` exists, the module `baz/index.js` will be returned as the module requested.  `systemjs-builder` adds this module as `foo/bar/baz` however, which screws up any dependency modules with relative pathing in `baz/index.js`.

`jspm` "solves" this issue by adding CommonJS JavaScript files for every directory like this, in order to patch this resolution jump.  For example, in the `foo/bar/baz/index.js` example, a file `for/bar/baz.js` is added, which has a single line: `module.exports = require("./baz/index.js")`, which allows resolution to continue normally.

If I was allowed to ask for a change / offer a suggestion for `systemjs-builder`:

When resolving a CommonJS path AND the path is a directory AND there exists an `index.js` file in the directory, create a module in memory of the form:

```
System.registerDynamic("foo/bar/baz.js", ["./baz/index.js"], function(...){ /* pass through code *});
```
and add this intermediary, single line module to the resulting bundle. This would allow `systemjs-builder` to resolve CommonJS bundles defined by npm's standard without the use of jspm (all other benefits of `jspm` ignored).

Thanks for your help.
